### ISSUE TITLE
Error: Failed to parse python binary; Reason: No such file or directory (os error 2)

### ISSUE THREAD

**Issue by yarikoptic** [benfred/py-spy#360]
well, my setup is quite obscure (described below) but I think it should have worked.  And if it isn't supposed to - would have been nice if error message was more informative: which binary it failed to parse and/or which file/directory is missing (I do not see some `--debug` or `--verbose` in the output of `py-spy [top] --help` which could optionally start giving me more information).

Setup: a singularity container in which I am running a python process using system wide (debian) python3.

```
(py-spy) bids@rolando:~$ export | grep SING
declare -x SINGULARITY_CONTAINER="reproin_0.9.0.simg"
declare -x SINGULARITY_NAME="reproin_0.9.0.simg"

```

 Since container environment is "fixed" and I have not pre-installed py-spy into it, I have initiated python virtualenv in it, using the same python3:
```
python3 -m virtualenv --system-site-packages --python python3 /tmp/py-spy
```

which I activated and then `pip install py-spy` into it.  Trying to run `py-spy top` though results in
```
(py-spy) bids@rolando:~$ py-spy top -p 39864
Error: Failed to parse python binary
Reason: No such file or directory (os error 2)

(py-spy) bids@rolando:~$ ls -l `which python`
lrwxrwxrwx 1 bids bids 7 Mar 15 17:36 /tmp/py-spy/bin/python -> python3
(py-spy) bids@rolando:~$ ls -l `which python3`
-rwxr-xr-x 1 bids bids 4861504 Mar 15 17:36 /tmp/py-spy/bin/python3
```

stracing it leads to

```
[pid 40252] open("/proc/39864/root/usr/local/singularity_v2.5.1/var/singularity/mnt/final/usr/bin/python3.7", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
```
so I guess it is the one it is trying to parse somehow instead of just 

```
(py-spy) bids@rolando:~$ ls -l /usr/bin/python3.7     
-rwxr-xr-x 2 root root 4861504 Jul 25  2020 /usr/bin/python3.7
```

**Comment by Jongy** [comment]
My fix in #364 is relevant to 2 cases:
1. Deleted executables are accessible via `/proc/pid/exe` and not via `/proc/pid/root/...`
2. When dealing with cross-namespace, accessing through `/proc/pid/root` *may* yield some unexpected behavior when resolving symlinks. The thing is, if you have, for example, `/tmp -> /var/tmp` in your container, then `/proc/pid/root/tmp/` resolves to `/var/tmp` on the host and not on the container. Traversing through symlinks works as you'd expect (with regard to `/proc/pid/root`) only if they're relative, not absolute.

Your file isn't deleted so option 1 isn't relevant, I do see the target process is in a container... So give a try to master w/ #364 .

> (I do not see some --debug or --verbose in the output of py-spy [top] --help which could optionally start giving me more information).

Can use `RUST_LOG=debug py-spy top -p ....`

**Comment by benfred** [comment]
I'm also wondering if the fix from @Jongy (#364) will fix this out -  I've pushed out a new release with this fix included, which should be easier than trying to build from source here.

Can you install the latest version (```pip install py-spy==0.3.5```) and see if this fixes your problem?

**Comment by benfred** [comment]
Going to guess this is working now w/ 0.3.5 - feel free to re-open if this is still broken
